### DISCUSSION TITLE
Is it possible to generate or hard code values for optional fields?

### DISCUSSION THREAD

**Discussion by szakrewsky** [sngular/kloadgen#441] [Q&A]
I am having a hard time generating values for optional fields.  I have added some samples to the Fields Values List.  I have also tried string and it types.  For optional fields (union with nulls) kloadgen avro is always sending null.  I looked at the code and it looks to me that kloadgen will skip all optional fields unless they meet that last field needed for parent qualification thing mentioned in the wiki.  Can someone confirm or provide direction on how to make optional fields not send null?

**Comment by jemacineiras** [comment]
HI  @szakrewsky,
thanks for the question, and sorry for the late reply.
We do not skip optional fields, at least not intentionally, if so is a bug. What we have a logic which randomly will fill or not those optional fields. But in the case all fields are optional, and the object must be generated because is mark as required, we will generate the latest field. If it's not working, then is a bug

**Comment by szakrewsky** [comment]
I spent some more time on this, but since it is through my employer, it is unlikely I will be able to contribute a fix.

I do have a work around.

This is with kloadgen version 5.6.3 and jmeter 5.6.3.

The symptoms of this is that even though you set everything up to send a value, optional values will always be null.  Optional values are UNIONS with null as a type.

In this case the work around is:
1. Open up the jmx file.
2. Find the field in question.
3. There will be a property `<boolProp name="required">false</boolProp>`.
4. Change this to true.
5. Reload the file in jmeter and it will work as expected.
